validator/tests/unit3/test_client_request_handlers/tests.py

```diff
@@ -0,0 +1,410 @@
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import unittest
+
+import sawtooth_validator.state.client_handlers as handlers
+from sawtooth_validator.protobuf import client_pb2
+from sawtooth_validator.protobuf.block_pb2 import Block
+from test_client_request_handlers.mocks import MockBlockStore
+from test_client_request_handlers.mocks import make_db_and_store
+
+
+class _ClientHandlerTestCase(unittest.TestCase):
+    """
+    Parent class for Client Request Handler tests that handles making requests.
+    Run _initialize as part of setUp, and then call _make_request in each test.
+    """
+    def _initialize(self, handler, request_proto, response_proto,
+                    store=None, roots=None):
+        self._identity = '1234567'
+        self._handler = handler
+        self._status = response_proto
+        self._request_proto = request_proto
+        self._store = store
+        self._roots = roots
+
+    def _make_request(self, **kwargs):
+        return self._handle(self._serialize(**kwargs))
+
+    def _serialize(self, **kwargs):
+        request = self._request_proto(**kwargs)
+        return request.SerializeToString()
+
+    def _handle(self, request):
+        result = self._handler.handle(self._identity, request)
+        return result.message_out
+
+    def _make_bad_request(self, **kwargs):
+        return self._handle(self._serialize(**kwargs)[0:-1])
+
+    def _break_genesis(self):
+        self._store.store.pop('chain_head_id')
+
+
+class TestStateCurrentRequests(_ClientHandlerTestCase):
+    def setUp(self):
+        def mock_current_root():
+            return '123'
+
+        self._initialize(
+            handlers.StateCurrentRequest(mock_current_root),
+            client_pb2.ClientStateCurrentRequest,
+            client_pb2.ClientStateCurrentResponse)
+
+    def test_state_current_request(self):
+        """Verifies requests for the current merkle root work properly.
+        Compares response to a mock merkle root of '123'.
+        """
+        response = self._make_request()
+
+        self.assertEqual(self._status.OK, response.status)
+        self.assertEqual('123', response.merkle_root)
+
+
+class TestStateListRequests(_ClientHandlerTestCase):
+    def setUp(self):
+        db, store, roots = make_db_and_store()
+        self._initialize(
+            handlers.StateListRequest(db, store),
+            client_pb2.ClientStateListRequest,
+            client_pb2.ClientStateListResponse,
+            store=store,
+            roots=roots)
+
+    def test_state_list_request(self):
+        """Verifies requests for data lists without parameters work properly.
+        Checks status, head_id, and length and type of leaves list.
+        """
+        response = self._make_request()
+
+        self.assertEqual(self._status.OK, response.status)
+        self.assertEqual('2', response.head_id)
```

Comment:
Additional comments in the API doc explaining the magic values here ('2' and 3) would help (why should it be '2' and 3 at this point?).  (Comment really applies to several of the tests here that test 'magic' values.)